File: back/Back/Domains/DailyWeather.cs
using System.Text.Json.Serialization;

namespace Back.Domains
{
    public class DailyWeather
    {
        [JsonInclude]
        private List<DateTime> Time { get; set; } = new List<DateTime>();
        [JsonPropertyName("weather_code")]
        [JsonInclude]
        private List<int> WeatherCode { get; set; } = new List<int>();  
        [JsonPropertyName("temperature_2m_max")]
        [JsonInclude]
        private List<double> Temperature2mMax { get; set; } = new List<double>();
        [JsonPropertyName("temperature_2m_min")]
        [JsonInclude]
        private List<double> Temperature2mMin { get; set; } = new List<double>();
        [JsonInclude]
        private List<DateTime> Sunrise { get; set; } = new List<DateTime>();
        [JsonInclude]
        private List<DateTime> Sunset { get; set; } = new List<DateTime>();
        [JsonPropertyName("daylight_duration")]
        [JsonInclude]
        private List<double> DaylightDuration { get; set; } = new List<double>();
        [JsonPropertyName("uv_index_max")]
        [JsonInclude]
        private List<double> UvIndexMax { get; set; } = new List<double>();
        [JsonPropertyName("precipitation_hours")]
        [JsonInclude]
        private List<double> PrecipitationHours { get; set; } = new List<double>();
        [JsonPropertyName("precipitation_probability_max")]
        [JsonInclude]
        private List<int> PrecipitationProbabilityMax { get; set; } = new List<int>();
        [JsonPropertyName("wind_speed_10m_max")]
        [JsonInclude]
        private List<double> WindSpeed10mMax { get; set; } = new List<double>();

        public DailyWeather()
        {
        }

        public DailyWeather(List<DateTime> time, List<int> weatherCode, List<double> temperature2mMax, List<double> temperature2mMin, List<DateTime> sunrise, List<DateTime> sunset, List<double> daylightDuration, List<double> uvIndexMax, List<double> precipitationHours, List<int> precipitationProbabilityMax, List<double> windSpeed10mMax)
        {
            Time = time;
            WeatherCode = weatherCode;
            Temperature2mMax = temperature2mMax;
            Temperature2mMin = temperature2mMin;
            Sunrise = sunrise;
            Sunset = sunset;
            DaylightDuration = daylightDuration;
            UvIndexMax = uvIndexMax;
            PrecipitationHours = precipitationHours;
            PrecipitationProbabilityMax = precipitationProbabilityMax;
            WindSpeed10mMax = windSpeed10mMax;
        }
    }
}

File: front/models/DailyWeather.ts
export interface IDailyWeather {
    time?: Date[];
    weatherCode?: number[];
    temperature2mMax?: number[];
    temperature2mMin?: number[];
    sunrise?: Date[];
    sunset?: Date[];
    daylightDuration?: number[];
    uvIndexMax?: number[];
    precipitationHours?: number[];
    precipitationProbabilityMax?: number[];
    windSpeed10mMax?: number[];
  }
  
  export class DailyWeather {
    private _time: Date[];
    private _weatherCode: number[];
    private _temperature2mMax: number[];
    private _temperature2mMin: number[];
    private _sunrise: Date[];
    private _sunset: Date[];
    private _daylightDuration: number[];
    private _uvIndexMax: number[];
    private _precipitationHours: number[];
    private _precipitationProbabilityMax: number[];
    private _windSpeed10mMax: number[];
  
    constructor({
      time = [],
      weatherCode = [],
      temperature2mMax = [],
      temperature2mMin = [],
      sunrise = [],
      sunset = [],
      daylightDuration = [],
      uvIndexMax = [],
      precipitationHours = [],
      precipitationProbabilityMax = [],
      windSpeed10mMax = [],
    }: IDailyWeather) {
      this._time = time;
      this._weatherCode = weatherCode;
      this._temperature2mMax = temperature2mMax;
      this._temperature2mMin = temperature2mMin;
      this._sunrise = sunrise;
      this._sunset = sunset;
      this._daylightDuration = daylightDuration;
      this._uvIndexMax = uvIndexMax;
      this._precipitationHours = precipitationHours;
      this._precipitationProbabilityMax = precipitationProbabilityMax;
      this._windSpeed10mMax = windSpeed10mMax;
    }
  
    get time(): Date[] {
      return this._time;
    }
  
    set time(value: Date[]) {
      this._time = value;
    }
  
    get weatherCode(): number[] {
      return this._weatherCode;
    }
  
    set weatherCode(value: number[]) {
      this._weatherCode = value;
    }
  
    get temperature2mMax(): number[] {
      return this._temperature2mMax;
    }
  
    set temperature2mMax(value: number[]) {
      this._temperature2mMax = value;
    }
  
    get temperature2mMin(): number[] {
      return this._temperature2mMin;
    }
  
    set temperature2mMin(value: number[]) {
      this._temperature2mMin = value;
    }
  
    get sunrise(): Date[] {
      return this._sunrise;
    }
  
    set sunrise(value: Date[]) {
      this._sunrise = value;
    }
  
    get sunset(): Date[] {
      return this._sunset;
    }
  
    set sunset(value: Date[]) {
      this._sunset = value;
    }
  
    get daylightDuration(): number[] {
      return this._daylightDuration;
    }
  
    set daylightDuration(value: number[]) {
      this._daylightDuration = value;
    }
  
    get uvIndexMax(): number[] {
      return this._uvIndexMax;
    }
  
    set uvIndexMax(value: number[]) {
      this._uvIndexMax = value;
    }
  
    get precipitationHours(): number[] {
      return this._precipitationHours;
    }
  
    set precipitationHours(value: number[]) {
      this._precipitationHours = value;
    }
  
    get precipitationProbabilityMax(): number[] {
      return this._precipitationProbabilityMax;
    }
  
    set precipitationProbabilityMax(value: number[]) {
      this._precipitationProbabilityMax = value;
    }
  
    get windSpeed10mMax(): number[] {
      return this._windSpeed10mMax;
    }
  
    set windSpeed10mMax(value: number[]) {
      this._windSpeed10mMax = value;
    }

    toJSON() {
        return {
          time: this._time.map(date => date.toISOString()),
          weatherCode: this._weatherCode,
          temperature2mMax: this._temperature2mMax,
          temperature2mMin: this._temperature2mMin,
          sunrise: this._sunrise.map(date => date.toISOString()),
          sunset: this._sunset.map(date => date.toISOString()),
          daylightDuration: this._daylightDuration,
          uvIndexMax: this._uvIndexMax,
          precipitationHours: this._precipitationHours,
          precipitationProbabilityMax: this._precipitationProbabilityMax,
          windSpeed10mMax: this._windSpeed10mMax,
        };
      }
    
      static fromJSON(data: IDailyWeather): DailyWeather {
        return new DailyWeather({
          time: data.time?.map(date => new Date(date)),
          weatherCode: data.weatherCode,
          temperature2mMax: data.temperature2mMax,
          temperature2mMin: data.temperature2mMin,
          sunrise: data.sunrise?.map(date => new Date(date)),
          sunset: data.sunset?.map(date => new Date(date)),
          daylightDuration: data.daylightDuration,
          uvIndexMax: data.uvIndexMax,
          precipitationHours: data.precipitationHours,
          precipitationProbabilityMax: data.precipitationProbabilityMax,
          windSpeed10mMax: data.windSpeed10mMax,
        });
      }
  }
  